src/component/reducer/helper/Zoom1DManager.ts

```diff
@@ -14,6 +14,10 @@ export const ZOOM_TYPES = {
 
 export type ZoomType = keyof typeof ZOOM_TYPES;
 
+function toFixed(value: number) {
+  return Number(value.toFixed(7));
```

Comment:
There's no guarantee that it will stay with this precision when converted back to a number.
What problem does this PR attempt to fix?

Reply:
I found if the value fraction is too small (1e-7 ... etc) this makes a problem in the plotting and the spectrum is no longer visible in the other hand it does not make sense to zoom to infinity.

Reply:
Converting to a string is certainly not the way to fix this issue.

Reply:
we should first put a limitation( hard code value) on the max value for the domain to prevent the infinity zoom but it still does not know what should be the value and based on what

Reply:
The problem that Hamed tries to fix is that if you scale too much the spectrum disappear.

![2023-07-22 17 06 44](https://github.com/cheminfo/nmrium/assets/1484241/1c2ff92b-e6d7-4016-b847-659c4efa56f6)

The reason for me is not clear. what are the values of the array when it disappear ? Is there a NaN in the array or an infinity ?

I would expect a 'Math.min' or 'Math.max' somewhere to fix the issue if it is related to an infinity or NaN.

Reply:
I still need to do more testing to see exactly when this bug happen and why and this why and this why i set the status of the PR as pending

Reply:
we limit the vertical zoom-in based on the spectra max value and multiply it with 0.0001

Reply:
Why we do that because if we zoom in for infinity some of the scaled values it too big or too small that the path element cannot display it, for example `-2.12222222e+10` so i set limitation on the zoom in